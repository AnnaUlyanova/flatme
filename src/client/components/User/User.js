import React, { PropTypes } from 'react'

const User = React.createClass({
  componentDidMount () {
    this.props.fetchUser()
  },
  render () {
    const user = this.props.user
    const title = user ? `${user.firstName} ${user.lastName}` : null
    return (
      <div>
        <h1>{title}</h1>
      </div>
    )
  }
})

User.propTypes = {
  fetchUser: PropTypes.func.isRequired,
  user: PropTypes.shape({
    firstName: PropTypes.string.isRequired,
    lastName: PropTypes.string.isRequired
  })
}

export default User
