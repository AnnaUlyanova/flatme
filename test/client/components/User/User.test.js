/* eslint-env jest */

import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import React from 'react'

import User from '../../../../src/client/components/User'

test('User renders correctly', () => {
  const fetchUser = () => {}
  const wrapper = shallow(<User fetchUser={fetchUser} />)
  expect(toJSON(wrapper)).toMatchSnapshot()
})
