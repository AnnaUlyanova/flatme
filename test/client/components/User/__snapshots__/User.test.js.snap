exports[`test User renders correctly 1`] = `
<div>
  <h1 />
</div>
`;
